### PR TITLE
another round of nanosecond conversions

### PR DESCRIPTION
**Pull request by Dudemanguy**
This should be all of it now. The most dangerous conversion here is, of course, the audio ones. I'm only actually able to test jack and pipewire which work fine as far as I can tell. Anyways, I'll let this sit for a while so errors and so on can be spotted (found a couple while opening this PR).

### CODE REVIEW

File: audio/out/ao_coreaudio.c [hunk 1]
```diff
@@ -53,13 +53,13 @@ static int64_t ca_get_hardware_latency(struct ao *ao) {
...
 
-    uint64_t audiounit_latency_us = audiounit_latency_sec * 1e6;
-    uint64_t device_latency_us    = ca_get_device_latency_us(ao, p->device);
+    uint64_t audiounit_latency_ns = MP_TIME_S_TO_NS(audiounit_latency_sec);
+    uint64_t device_latency_ns    = ca_get_device_latency_ns(ao, p->device);
 
-    MP_VERBOSE(ao, "audiounit latency [us]: %lld\n", audiounit_latency_us);
-    MP_VERBOSE(ao, "device latency [us]: %lld\n", device_latency_us);
+    MP_VERBOSE(ao, "audiounit latency [us]: %lld\n", audiounit_latency_ns);
+    MP_VERBOSE(ao, "device latency [us]: %lld\n", device_latency_ns);
```
Comment (orion1vi) [inline audio/out/ao_coreaudio.c:60]: [ns]

Comment (Dudemanguy) [reply audio/out/ao_coreaudio.c:60]: Always something. Thanks.

**Review by orion1vi — CHANGES_REQUESTED** [review]

File: audio/out/ao_coreaudio_utils.c [hunk 2]
```diff
@@ -292,9 +292,9 @@ bool ca_asbd_is_better(AudioStreamBasicDescription *req,
     return true;
 }
 
-int64_t ca_frames_to_us(struct ao *ao, uint32_t frames)
+int64_t ca_frames_to_ns(struct ao *ao, uint32_t frames)
 {
-    return frames / (float) ao->samplerate * 1e6;
+    return frames / MP_TIME_S_TO_NS(ao->samplerate);
```
[the remark this answers is not in the document]
Comment (Dudemanguy) [reply audio/out/ao_coreaudio_utils.c:297]: This and the other one return `in64_t` anyways. Also I'm not sure why this ever casted to float seeing as 1e6 should be a double.

[the remark this answers is not in the document]
Comment (orion1vi) [reply audio/out/ao_coreaudio_utils.c:297]: Current change will return 0, look at ao_audiotrack change if you're misunderstanding what I'm suggesting here.

[the remark this answers is not in the document]
Comment (Dudemanguy) [reply audio/out/ao_coreaudio_utils.c:297]: Oh my bad. Order of operations duh.

File: audio/out/ao_jack.c [hunk 3]
```diff
@@ -122,8 +122,8 @@ static int process(jack_nframes_t nframes, void *arg)
     jack_nframes_t jack_latency =
         atomic_load(&p->graph_latency_max) + atomic_load(&p->buffer_size);
 
-    int64_t end_time = mp_time_us();
-    end_time += (jack_latency + nframes) / (double)ao->samplerate * 1000000.0;
+    int64_t end_time = mp_time_ns();
+    end_time += (jack_latency + nframes) / (double)MP_TIME_S_TO_NS(ao->samplerate);
```
Comment (orion1vi) [inline audio/out/ao_jack.c:126]: Another one of those

Comment (Dudemanguy) [reply audio/out/ao_jack.c]: Thanks again.

**Comment by Dudemanguy** [comment]
The past couple of PRs seem to have gone smoothly overall so hopefully this will be the same.

**Pull request merged**